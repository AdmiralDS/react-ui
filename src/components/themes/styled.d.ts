import 'styled-components';
import type { Theme } from './index';

interface IPalette {
  main: string;
  contrastText: string;
}

declare module 'styled-components' {
  export interface DefaultTheme extends Theme {
    description?: string;
  }
}
