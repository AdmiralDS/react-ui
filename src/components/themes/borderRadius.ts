import type { Shape } from '#src/components/themes/common';
export const ALL_BORDER_RADIUS_VALUES = [
  'Border radius 0',
  'Border radius 2',
  'Border radius 4',
  'Border radius 6',
  'Border radius 8',
  'Border radius 10',
] as const;

export function smallGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '2px';
    case 'Border radius 4':
      return '4px';
    case 'Border radius 6':
      return '4px';
    case 'Border radius 8':
      return '4px';
    case 'Border radius 10':
      return '4px';
    default:
      return '4px';
  }
}

export function mediumGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '2px';
    case 'Border radius 4':
      return '4px';
    case 'Border radius 6':
      return '6px';
    case 'Border radius 8':
      return '8px';
    case 'Border radius 10':
      return '10px';
    default:
      return '4px';
  }
}

export function largeGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '4px';
    case 'Border radius 4':
      return '8px';
    case 'Border radius 6':
      return '12px';
    case 'Border radius 8':
      return '16px';
    case 'Border radius 10':
      return '20px';
    default:
      return '8px';
  }
}
