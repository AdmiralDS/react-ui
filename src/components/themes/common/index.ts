import { Color } from '#src/components/themes/common/color';
import { ZIndex } from '#src/components/themes/common/zIndex';
import baseStyled, { ThemedStyledInterface } from 'styled-components';

export * from './color';

export interface Shadow {
  ClickableActive: string;
  ClickableDefault: string;
  ClickableHover: string;
  NonClickable: string;
}

export interface Theme {
  color: Color;
  shadow: Shadow;
  zIndex: ZIndex;
  name: 'light' | 'dark';
}

export const styled = baseStyled as ThemedStyledInterface<Theme>;
