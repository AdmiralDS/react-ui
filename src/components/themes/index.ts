export * from './light';
export * from './dark';
export type { Theme } from './common';
