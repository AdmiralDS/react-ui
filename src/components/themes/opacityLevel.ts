export const OPACITY_LEVEL_1 = 0.02;
export const OPACITY_LEVEL_2 = 0.04;
export const OPACITY_LEVEL_3 = 0.08;
export const OPACITY_LEVEL_4 = 0.1;
export const OPACITY_LEVEL_5 = 0.15;
export const OPACITY_LEVEL_6 = 0.16;
export const OPACITY_LEVEL_7 = 0.2;
export const OPACITY_LEVEL_8 = 0.25;
export const OPACITY_LEVEL_9 = 0.3;
export const OPACITY_LEVEL_10 = 0.35;
export const OPACITY_LEVEL_11 = 0.4;
export const OPACITY_LEVEL_12 = 0.5;
export const OPACITY_LEVEL_13 = 0.6;
export const OPACITY_LEVEL_14 = 0.8;
export const OPACITY_LEVEL_15 = 0.95;
export const OPACITY_MAX_LEVEL = 1;
