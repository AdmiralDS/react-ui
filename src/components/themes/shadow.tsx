import { css } from 'styled-components';

export const shadowClickableActiveMixin = css`
  box-shadow: 0 2px 4px rgba(138, 150, 168, 0.4), 0px -2px 2px rgba(138, 150, 168, 0.08);
`;

export const shadowClickableDefaultMixin = css`
  box-shadow: 0 4px 8px rgba(138, 150, 168, 0.4), 0px -2px 4px rgba(138, 150, 168, 0.08);
`;

export const shadowNonClickableMixin = css`
  box-shadow: 0 4px 16px rgba(138, 150, 168, 0.4), 0 0 6px rgba(138, 150, 168, 0.08);
`;

export const shadowClickableHoverMixin = css`
  box-shadow: 0 8px 16px rgba(138, 150, 168, 0.4), 0px -2px 8px rgba(138, 150, 168, 0.08);
`;
