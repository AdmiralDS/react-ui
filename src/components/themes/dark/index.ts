import type { Theme } from '../common';
import { COLOR as color } from './color';
import { shadow } from './shadow';
import { zIndex } from '../common/zIndex';

export const DARK_THEME: Theme = {
  color,
  shadow,
  zIndex,
  name: 'dark',
};
