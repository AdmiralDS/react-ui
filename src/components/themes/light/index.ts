import type { Theme } from '../common';
import { COLOR as color } from './color';
import { shadow } from './shadow';
import { zIndex } from '../common/zIndex';

export const LIGHT_THEME = {
  color,
  shadow,
  zIndex,
  name: 'light',
} as Theme;
