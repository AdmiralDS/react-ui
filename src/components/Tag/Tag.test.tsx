import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { LIGHT_THEME } from '../themes';

import { Tag } from './';

describe('Tag', () => {
  const clickProps = {
    onClick: jest.fn(),
  };
  it('should render default component', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Tag>test</Tag>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with kind "green" and determined width', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Tag kind="green" width="60px">
          test
        </Tag>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when user clicks on component', () => {
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <Tag {...clickProps}>text</Tag>
      </ThemeProvider>,
    );
    fireEvent.click(wrapper.getByText('text'));
    const { onClick } = clickProps;
    expect(onClick).toBeCalledTimes(1);
  });
});
