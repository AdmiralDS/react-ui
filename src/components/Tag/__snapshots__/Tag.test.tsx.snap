// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render component with kind "green" and determined width 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  box-sizing: border-box;
  height: 24px;
  width: 60px;
  margin-top: 8px;
  padding: 3px 7px;
  border-radius: 4px;
  background: #E2E5E9;
  border: 1px solid #E2E5E9;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  pointer-events: none;
}

.c0:hover,
.c0:active {
  background: #D3D8DE;
}

.c0:focus {
  outline: none;
}

.c0:focus:before {
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border-radius: 6px;
  border: 2px solid #0A54C2;
}

.c2 {
  color: #2B313B;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin-right: 6px;
  background: #2ABB5B;
}

<body>
    <div>
      <button
        class="c0"
        width="60px"
      >
        <div
          class="c1"
        />
        <span
          class="c2"
        >
          test
        </span>
      </button>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  box-sizing: border-box;
  height: 24px;
  width: 60px;
  margin-top: 8px;
  padding: 3px 7px;
  border-radius: 4px;
  background: #E2E5E9;
  border: 1px solid #E2E5E9;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  pointer-events: none;
}

.c0:hover,
.c0:active {
  background: #D3D8DE;
}

.c0:focus {
  outline: none;
}

.c0:focus:before {
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border-radius: 6px;
  border: 2px solid #0A54C2;
}

.c2 {
  color: #2B313B;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  margin-right: 6px;
  background: #2ABB5B;
}

<div>
    <button
      class="c0"
      width="60px"
    >
      <div
        class="c1"
      />
      <span
        class="c2"
      >
        test
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tag should render default component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  position: relative;
  box-sizing: border-box;
  height: 24px;
  margin-top: 8px;
  padding: 3px 7px;
  border-radius: 4px;
  background: #E2E5E9;
  border: 1px solid #E2E5E9;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  pointer-events: none;
}

.c0:hover,
.c0:active {
  background: #D3D8DE;
}

.c0:focus {
  outline: none;
}

.c0:focus:before {
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border-radius: 6px;
  border: 2px solid #0A54C2;
}

.c1 {
  color: #2B313B;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

<body>
    <div>
      <button
        class="c0"
      >
        <span
          class="c1"
        >
          test
        </span>
      </button>
    </div>
  </body>,
  "container": .c0 {
  position: relative;
  box-sizing: border-box;
  height: 24px;
  margin-top: 8px;
  padding: 3px 7px;
  border-radius: 4px;
  background: #E2E5E9;
  border: 1px solid #E2E5E9;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  pointer-events: none;
}

.c0:hover,
.c0:active {
  background: #D3D8DE;
}

.c0:focus {
  outline: none;
}

.c0:focus:before {
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border-radius: 6px;
  border: 2px solid #0A54C2;
}

.c1 {
  color: #2B313B;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 12px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
}

<div>
    <button
      class="c0"
    >
      <span
        class="c1"
      >
        test
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
